src/preprocess/preprocess.py

```diff
@@ -49,66 +72,117 @@ def remove_all(list_uid, f):
+                                anonimize_coordinate, departament[key][dataset_type], f)
+                    if has_tracker_program:
+                        generate_anonymize_tracker_rules(tracker_program, tracker_entity_attributes, org_units, data_elements,
+                                anonimize_org_units, anonimize_phone, anonimize_mail,
+                                anonimize_coordinate,
+                                                         departament[key][tracker_type], f)
+                    if has_event_program:
+                        generate_anonymize_event_rules(event_program, org_units, data_elements,
+                                anonimize_org_units, anonimize_phone, anonimize_mail,
+                                anonimize_coordinate,
+                                                       departament[key][program_type], f)
+                    # if not have specific rules apply to all the departament
+                    if not has_datasets and not has_event_program and not has_tracker_program:
+                        for key_type in departament[key].keys():
+                            if key_type == program_type:
+                                generate_anonymize_event_rules(event_program, org_units, data_elements,
+                                                               anonimize_org_units, anonimize_phone, anonimize_mail,
+                                                               anonimize_coordinate,
+                                                               departament[key][program_type], f)
+                            if key_type == dataset_type:
+                                generate_anonymize_datasets_rules(datasets, org_units, data_elements,
+                                anonimize_org_units, anonimize_phone, anonimize_mail,
+                                anonimize_coordinate, departament[key][dataset_type], f)
+                            if key_type == tracker_type:
+                                generate_anonymize_tracker_rules(tracker_program, tracker_entity_attributes,
+                                                                 org_units, data_elements,
+                                                                 anonimize_org_units, anonimize_phone, anonimize_mail,
+                                                                 anonimize_coordinate,
+                                                                 departament[key][tracker_type], f)
+
+
+def get_rule_content(rule, rule_type):
+    if rule_type in rule.keys():
+        return rule[rule_type]
+    if rule_type in [anonymize_phone_rule, anonymize_coordinates_rule, anonymize_mail_rule]:
+        return True
+    elif rule_type == anonymize_org_unit_rule:
+        return False
+    else:
+        return ""
+
+
+def check_if_has_metadata_type(rule, check_type):
+    for rule_type in rule[metadata_type]:
+        if rule_type == check_type:
+            return True
+    return False
```

Comment:
[minor] Check https://docs.python.org/3/library/functions.html#any

Reply:
mm that code is not evaluating if the rule_type exist, is evaluating if the check_type exist in the metadata_type rule. 
But i think the actual code is wrong i have changed it. I need test it before ask for a new review